backend/baseclass.pm

```diff
@@ -1204,14 +1204,15 @@ sub start_ssh_serial {
 
     $self->stop_ssh_serial;
 
-    $self->{serial} = $self->new_ssh_connection(%args);
-    my $chan = $self->{serial}->channel();
-    die "No channel found" unless $chan;
-    $self->{serial_chan} = $chan;
+    my $ssh  = $self->{serial}      = $self->new_ssh_connection(%args);
+    my $chan = $self->{serial_chan} = $ssh->channel();
+    if (!$chan) {
+        $ssh->die_with_error("Unable to establish SSH channel for serial console");
+    }
     $chan->blocking(0);
     $chan->pty(1);
     $self->{select}->add($self->{serial}->sock);
-    return $chan;
+    return ($ssh, $chan);
```

Comment:
I guess you probably conflict with https://github.com/os-autoinst/os-autoinst/pull/1109. Good idea, but we both (+ @okurz did as well) are working on SSH code refactoring (addressing similar issues, but not quite the same). After rebasing https://github.com/os-autoinst/os-autoinst/pull/1113 I hope I won't need to rebase this one as well :).

BTW, during https://github.com/os-autoinst/openQA/pull/2003 I wasn't sure, whether returning `$ssh` wouldn't be enough in functions creating SSH connection. I mean anybody can call `$ssh->channel()`, so why to complicate returning with list. But probably having a test on `$chan` make justify it. 
\+1 for using `die_with_error()` from `Net::SSH2`, I was also thinking about it.

Reply:
This *must* return `$ssh` to be useful. Otherwise gathering error information e.g. in case of a failed `$chan->exec` is not possible. I can help rebasing if required.

Note that this PR is supposed to supersede the one created by @okurz.

Reply:
Sorry, I meant return *only* `$ssh` (without `$chan`), sure `$ssh` should returned. My other concerns is speed (is it `$chan` a new connection or reusing already existed one; because speed is one of 2 benefits for serial console) + whether we need sometimes other SSH instance than `$self->{ssh}`).

> Note that this PR is supposed to supersede the one created by @okurz.

I'm changing that code differently in https://github.com/os-autoinst/os-autoinst/pull/1109 :(. Any objections?

Reply:
Feel free to change it to return only `$ssh`.

Reply:
If you don't require it, I'd prefer to keep it as it is in order to have it merged and address these issues later. This PR does quite a lot of cleanup already :).